modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache

```diff
@@ -51,7 +52,14 @@ export function {{classname}}ToJSON(value?: {{classname}} | null): any {
             throw new Error(`No variant of {{classname}} exists with '{{discriminator.propertyName}}=${value['{{discriminator.propertyName}}']}'`);
     }
 {{/discriminator}}
+
 {{^discriminator}}
-    return { {{#oneOf}}...{{{.}}}ToJSON(value){{^-last}}, {{/-last}}{{/oneOf}} };
+    {{#oneOf}}
+    if (instanceOf{{{.}}}(value)) {
+        return {{{.}}}ToJSON(value as {{{.}}});
+    }
+    {{/oneOf}}
+
+    return {};
 {{/discriminator}}
```

Comment:
I'm not entirely familiar with this code, but I think something similar should also be added to `modelGeneric.mustache`.
In a project of mine where I use `allOf` for inheritance. The base class has a `{class}FromJSONTyped` function that handles all the discriminator cases, but `{class}ToJSON` doesn't include any discriminator code at all and just returns an object with properties of the base class.

I'm currently patching the generated code to add `ignoreDiscriminator` to the `ToJSON` functions which will be set to `true` when invoked from the child class's `ToJSON`, if `false` `ToJSON` in the base class calls the respective child class `ToJSON` based on the discriminator.

Here are the relevant models of my openapi config:
```json
      "Activity": {
        "required": ["$type", "id", "title"],
        "type": "object",
        "properties": {
          "$type": {
            "$ref": "#/components/schemas/ActivityType"
          },
          "id": {
            "type": "string"
          },
          "title": {
            "type": "string"
          },
        },
        "additionalProperties": false,
        "discriminator": {
          "propertyName": "$type",
          "mapping": {
            "exercise": "#/components/schemas/ActivityExercise",
            "job": "#/components/schemas/ActivityJob",
          }
        }
      },
      "ActivityExercise": {
        "required": ["$type", "category"],
        "type": "object",
        "allOf": [
          {
            "$ref": "#/components/schemas/Activity"
          }
        ],
        "properties": {
          "$type": {
            "$ref": "#/components/schemas/ActivityType"
          },
          "category": {
            "$ref": "#/components/schemas/ActivityCategory"
          },
          "muscleShares": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/MuscleShare"
            },
            "nullable": true
          }
        },
        "additionalProperties": false
      },
      "ActivityJob": {
        "required": ["$type"],
        "type": "object",
        "allOf": [
          {
            "$ref": "#/components/schemas/Activity"
          }
        ],
        "properties": {
          "$type": {
            "$ref": "#/components/schemas/ActivityType"
          },
          "muscleShares": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/MuscleShare"
            },
            "nullable": true
          }
        },
        "additionalProperties": false
      },
```